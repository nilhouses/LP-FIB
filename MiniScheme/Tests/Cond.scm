;Aquest test es pot executar des del directori arrel del projecte amb les comandes:

  ;SI VOLEM   COMPROVAR EL CORRECTE FUNCIONAMENT DE LA GRAMÃ€TICA
    ;python3 scheme.py ./Tests/Cond.scm | diff - ./Tests/Cond.out

  ;SI VOLEM VEURE LA SORTIDA PER PANTALLA
    ;python3 scheme.py ./Tests/Cond.scm


(define x 3)
(define y 5)
(define true #t)

(define (main)
  (display "Comprovem que un cond funcioni correctament en 3 casos")
  (display "3 es ___ que 5: ")
  (newline)
  (display
      (cond
        ((> 3 5) "major")
        (true "menor")
        (#t "igual"))
    )
  (newline)
  (display
    (cond
      ((> 3 5) "major")
      ((= 3 5) "igual")
      (else "menor"))
  )
  (newline)
  (display
      (cond
        ((< x y) "menor")
        ((= x y) "igual")
        (true "major"))
    )
  (newline)
)
