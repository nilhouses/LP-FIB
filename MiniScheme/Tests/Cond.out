Comprovem que un cond funcioni correctament en 3 casos3 es ___ que 5: 
menor
menor
menor
